client/state/shortcodes/reducer.js

```diff
@@ -50,6 +52,32 @@ export const requesting = createReducer( {}, {
  * @return {Object}        Updated state
  */
 export const items = createReducer( {}, {
+	[ MEDIA_ITEMS_RECEIVE ]: ( state, { siteId, data } ) => {
+		if ( ! state.hasOwnProperty( siteId ) ) {
+			return state;
+		}
+
+		let media;
+		if ( Array.isArray( data.media ) ) {
```

Comment:
Would be better if the data was already normalized to our expected form by the time it reaches the reducer, either from the action creator or Flux compatibility layer.